src/apps/competitions/tests/test_phase_migration.py

```diff
@@ -0,0 +1,106 @@
+    CompetitionParticipantFactory
+
+twenty_minutes_ago = now() - datetime.timedelta(hours=0, minutes=20)
+five_minutes_ago = now() - datetime.timedelta(hours=0, minutes=5)
+
+
+class CompetitionPhaseToPhase(TestCase):
+    def setUp(self):
+        # Competition Owner
+        self.owner = UserFactory(username='owner', super_user=True)
+
+        # Users
+        self.normal_user = UserFactory(username='norm')
+        self.normal_user2 = UserFactory(username='norm2')
+        self.normal_user3 = UserFactory(username='norm3')
+
+        # Competitions
+        self.competition = CompetitionFactory(created_by=self.owner)
+        self.competition2 = CompetitionFactory(created_by=self.owner)
+
+        # Competition Participants
+        self.competition_participant = CompetitionParticipantFactory(user=self.normal_user,
+                                                                     competition=self.competition)
+        self.competition_participant2 = CompetitionParticipantFactory(user=self.normal_user2,
+                                                                      competition=self.competition)
+        self.competition_participant3 = CompetitionParticipantFactory(user=self.normal_user3,
+                                                                      competition=self.competition)
+
+        # Competition 1 Phases
+        self.phase1 = PhaseFactory(competition=self.competition, auto_migrate_to_this_phase=False,
+                                   end=twenty_minutes_ago, index=1, name='Phase1')
+        self.phase2 = PhaseFactory(competition=self.competition, auto_migrate_to_this_phase=True,
+                                   start=five_minutes_ago, index=2, name='Phase2')
+        self.phase3 = PhaseFactory(competition=self.competition, auto_migrate_to_this_phase=False,
+                                   index=3, name='Phase3')
+
+        # Competition 2 Phases
+        self.phase1_competition2 = PhaseFactory(competition=self.competition2, auto_migrate_to_this_phase=False,
+                                                end=twenty_minutes_ago, has_been_migrated=True,
+                                                index=1)
+        self.phase2_competition2 = PhaseFactory(competition=self.competition2, auto_migrate_to_this_phase=True,
+                                                start=five_minutes_ago, index=2)
+
+        # Competition 1 Submissions
+        self.submission = self.make_submission()
+        self.submission2 = self.make_submission(participant=self.competition_participant2)
+
+        # Competition 2 Submissions
+        self.submission1_competition2 = self.make_submission(phase=self.phase1_competition2,
+                                                             participant=self.competition_participant)
+        self.submission2_competition2 = self.make_submission(phase=self.phase1_competition2,
+                                                             participant=self.competition_participant3)
```

Comment:
Do we need all this setup for all of these tests? can we pare this down into only what we need for every test scenario and then supplement in the tests themselves?

Reply:
Removed a lot of unused setup and added a loop for submissions in the setup.